File: 序号/901.股票价格跨度.cpp
class StockSpanner {
  deque<pair<int, int>> q;
  int idx = -1;

 public:
  /** 输入当天价格，返回连续小于等于当前价格的天数 */
  StockSpanner() { q.emplace_back(-1, 10001); }

  /**  栈内只保留比当前更大的数，每次存数据时弹出更小的，计算索引差 */
  int next(int price) {
    idx++;
    while (q.back().second <= price) {
      q.pop_back();
    }
    int ans = idx-q.back().first;
    q.emplace_back(idx, price);
    return ans;
  }
};

File: 序号/901. 股票价格跨度.py
class StockSpanner:

    def __init__(self):
        self.q=deque()
        self.idx=-1
        self.q.append((-1,math.inf))

    def next(self, price: int) -> int:
        self.idx+=1
        while self.q[-1][1]<=price:
            self.q.pop()
        ans=self.idx-self.q[-1][0]
        self.q.append((self.idx,price))
        return ans


# Your StockSpanner object will be instantiated and called as such:
# obj = StockSpanner()
# param_1 = obj.next(price)